go-controller/pkg/clustermanager/status_manager/status_manager_test.go

```diff
@@ -126,6 +129,52 @@ func checkEmptyAPBRouteStatusConsistently(apbRoute *adminpolicybasedrouteapi.Adm
 	}).Should(BeTrue(), "expected Status to be consistently empty")
 }
 
+func newEgressQoS(namespace string) *egressqosapi.EgressQoS {
+	return &egressqosapi.EgressQoS{
+		ObjectMeta: util.NewObjectMeta("default", namespace),
+		Spec: egressqosapi.EgressQoSSpec{
+			Egress: []egressqosapi.EgressQoSRule{
+				{
+					DSCP:    60,
+					DstCIDR: pointer.String("1.2.3.4/32"),
+				},
+			},
+		},
+	}
+}
+
+func updateEgressQoSStatus(egressQoS *egressqosapi.EgressQoS, status *egressqosapi.EgressQoSStatus,
+	fakeClient *util.OVNClusterManagerClientset) {
+	egressQoS.Status = *status
+	_, err := fakeClient.EgressQoSClient.K8sV1().EgressQoSes(egressQoS.Namespace).
+		Update(context.TODO(), egressQoS, metav1.UpdateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func checkEQStatusEventually(egressQoS *egressqosapi.EgressQoS, expectFailure bool, expectEmpty bool, fakeClient *util.OVNClusterManagerClientset) {
+	Eventually(func() bool {
+		eq, err := fakeClient.EgressQoSClient.K8sV1().EgressQoSes(egressQoS.Namespace).
+			Get(context.TODO(), egressQoS.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if expectFailure {
+			return strings.Contains(eq.Status.Status, types.EgressQoSErrorMsg)
+		} else if expectEmpty {
+			return eq.Status.Status == ""
+		} else {
+			return strings.Contains(eq.Status.Status, "applied")
+		}
+	}).Should(BeTrue(), fmt.Sprintf("expected egress QoS status with expectFailure=%v expectEmpty=%v", expectFailure, expectEmpty))
+}
+
+func checkEmptyEQStatusConsistently(egressQoS *egressqosapi.EgressQoS, fakeClient *util.OVNClusterManagerClientset) {
+	Consistently(func() bool {
+		ef, err := fakeClient.EgressQoSClient.K8sV1().EgressQoSes(egressQoS.Namespace).
+			Get(context.TODO(), egressQoS.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		return ef.Status.Status == ""
+	}).Should(BeTrue(), "expected Status to be consistently empty")
+}
```

Comment:
heh I was about to ask you to write func descriptions for all these utils, but good naming scheme.. at least I am able to understand it, I hope this is also beginner friendly.

Reply:
We must thank @npinaeva for this, I just copied the code from EgressFirewall test logic :)